File: frontend/academia_ef/src/app/services/usuario.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { UsuarioInterface } from '../models/interfaces/usuario-interface';
import { UsuarioModel } from '../models/usuario-model';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {

  constructor(private http: HttpClient) { }

  configUrl = 'https://localhost:7170';

  login(login: string, senha: string): Observable<UsuarioInterface> {
    return this.http.get<UsuarioInterface>(`${this.configUrl}/Usuarios/Login/${login}/${senha}`);
  }

  getUsuarioSessao(): any{
    const usuario = {
      id: window.localStorage.getItem('id'),
      login: window.localStorage.getItem('login'),
      tipoDeUsuario: window.localStorage.getItem('tipoDeUsuario'),
    }

    return usuario;
  }

  logout(): any{
      window.localStorage.clear();
  }

  getStatusDaSessao() {
    return !!(window.localStorage.getItem('id') && 
    window.localStorage.getItem('login') && 
    window.localStorage.getItem('tipoDeUsuario'));
  }
}

File: backend/academia_ef/Services/UsuarioService.cs
using academia_ef.Model;
using academia_ef.Repository.Interfaces;
using academia_ef.Services.Interfaces;

namespace academia_ef.Services
{
    public class UsuarioService : IUsuarioService
    {
        public readonly IUsuarioRepository _usuarioRepository;

        public UsuarioService(IUsuarioRepository usuarioRepository)
        {
            _usuarioRepository = usuarioRepository;
        }

        public Usuario Login(string login, string senha)
        {
            return _usuarioRepository.Login(login, senha);
        }
    }
}
